pkg/deploy/api/types.go

```diff
@@ -77,17 +79,20 @@ type RecreateDeploymentStrategyParams struct {
 	// pod is created. All LifecycleHookFailurePolicy values are supported.
 	Mid *LifecycleHook
 	// Post is a lifecycle hook which is executed after the strategy has
-	// finished all deployment logic. The LifecycleHookFailurePolicyAbort policy
```

Comment:
In a failure case, the scaler reconciles the old back to the new - a failed
deployment gets rolled back by scaling.

On Thu, Feb 11, 2016 at 8:44 AM, David Eads <notifications@github.com>
wrote:

> In pkg/deploy/api/types.go
> <https://github.com/openshift/origin/pull/6925#discussion_r52602047>:
>
> > @@ -77,17 +79,20 @@ type RecreateDeploymentStrategyParams struct {
> >  	// pod is created. All LifecycleHookFailurePolicy values are supported.
> >  	Mid *LifecycleHook
> >  	// Post is a lifecycle hook which is executed after the strategy has
> > -	// finished all deployment logic. The LifecycleHookFailurePolicyAbort policy
>
> This change is really strange. Specifying Abort here doesn't seem to what
> I'd expect. The action has already happened, right? The new RC is up and
> running. I may have marked it as failed, but its actually there answering
> my requests, isn't it?
>
> I didn't think that we had compensation logic to handle an automatic
> rollback in the case of a deployment marked failure because it didn't
> matter before, you couldn't abort after the action was done.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6925/files#r52602047>.
>